instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java

```diff
@@ -66,18 +67,29 @@ public ContextPropagators getPropagators() {
     return propagators;
   }
 
-  public Span startSpan(Class<?> clazz) {
-    String spanName = spanNameForClass(clazz);
-    return startSpan(spanName, SpanKind.INTERNAL);
+  public Context startSpan(Class<?> clazz) {
+    return startSpan(spanNameForClass(clazz));
+  }
+
+  public Context startSpan(Method method) {
+    return startSpan(spanNameForMethod(method));
   }
 
-  public Span startSpan(Method method) {
-    String spanName = spanNameForMethod(method);
+  public Context startSpan(String spanName) {
```

Comment:
Hmm one high level comment - I think we have a dream of setting more attributes on `Span.Builder` where we can, currently we never do. Will we lose that with this sort of API? I guess if there's a `protected fillAttributes(Span.Builder)` on tracers, it should be ok just want to confirm.

Reply:
I think that all those `startSpan()` methods in the base tracer are not really meant for extension -- they're more like the simplest way of creating a span. Usually you need to pass something more than a string span name, or a `Class` to extract some useful attributes after all.
I think that we should explicitly state that in the javadoc: if you want something more than a bare span, write your own `startSpan()` method.